src/ralph/models/edx/base.py

```diff
@@ -0,0 +1,171 @@
+
+
+class BaseContextModel(BaseModelWithConfig):
+    """Represents the base context model inherited by all event contexts.
+
+    Attributes:
+        course_user_tags (dict of str: str): Content from `user_api_usercoursetag` table.
+            Retrieved with:
+                `dict(
+                    UserCourseTag.objects.filter(
+                        user=request.user.pk, course_id=course_key
+                    ).values_list('key', 'value')
+                )`
+            Note:
+                Is only present when a course page is requested.
+                Is an empty dictionary when the user is not logged in or not found in the
+                `user_api_usercoursetag` table.
+            Source:
+                /openedx/core/djangoapps/user_api/middleware.py#L38-L46
+        user_id (int or str or None): Consists of the ID of the authenticated user.
+            Retrieved with:
+                `request.user.pk` querying the `auth_user` table.
+            Note:
+                Is an integer when the user is found in the `auth_user` table.
+                Is an empty string when an exception is raised while retrieving the id.
+                Is `None` when the user is not logged in.
+            Source:
+                /common/djangoapps/track/middleware.py#L189
+        org_id (str): Consists of the organization name that lists the course.
+            Retrieved with:
+                `course_id.org` where `course_id` is an `opaque_keys.edx.locator.CourseLocator`
+                which is created using the URL of the requested page.
+            Note:
+                Is an empty string when the requested page is not a course page.
+            Source:
+                /common/djangoapps/track/contexts.py#L55
+        course_id (str): Consists of the unique identifier for the visited course page.
+            Retrieved with:
+                `course_id.to_deprecated_string()` where `course_id` is an
+                `opaque_keys.edx.locator.CourseLocator` which is created using the URL
+                of the requested page.
+            Note:
+                Is an empty string when the requested page is not a course page.
+            Source:
+                /common/djangoapps/track/contexts.py#L54
+        path (Path): Consist of the relative URL (without the hostname) of the requested page.
+            Retrieved with:
+                `request.META['PATH_INFO']`
+            Source:
+                /common/djangoapps/track/middleware.py#L143
+    """
+
+    course_user_tags: Optional[dict[str, str]]
+    user_id: Union[int, Literal[""], None]
+    org_id: str
+    course_id: str
+    path: Path
+
+    @root_validator
+    def validate_course_id(
+        cls, values
+    ):  # pylint: disable=no-self-argument, no-self-use
+        """The course_id must match the regex `course-v1:{org_id}\\+.+\\+.+` or an empty string."""
+
+        org_id, course_id = values.get("org_id"), values.get("course_id")
```

Comment:
Not a big fan of this shortener, I would prefer being explicit:
```suggestion
        org_id = values.get("org_id")
        course_id = values.get("course_id")
```